### PR TITLE
 JENKINS-61201 JEP-224 System read - about jenkins

### PR DESCRIPTION
**Pull request by timja**
See [JEP-224](https://jenkins.io/jep/224) https://github.com/jenkinsci/jenkins/pull/4506

Adds system read permission support to about jenkins

## Screenshots

Nothing changed in this PR in the UI

### Manual testing notes

When using the core-pr-tester (`docker run --rm -ti -p 8080:8080 -e ID=4520 jenkins/core-pr-tester`), you can use script console to enable the new permission: 

`Jenkins.SYSTEM_READ.enabled = true`

### Proposed changelog entries

* Entry 1: JENKINS-61201, Allows users with system read permission to view the about jenkins page
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests - can implement if someone wants...
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by daniel-beck** [comment]
Last I checked (for _Manage_) this was a huge mess because the links to `/thirdPartyLicenses` require _Administer_, and once that was fixed, we had to fix `PluginWrapper#getApi` so no exceptions are logged on that page.

Did you test this at all?

**Comment by timja** [comment]
I didn’t click any links on the page they all looked like they were going off site, will check closer

**Comment by timja** [comment]
> Last I checked (for _Manage_) this was a huge mess because the links to `/thirdPartyLicenses` require _Administer_, and once that was fixed, we had to fix `PluginWrapper#getApi` so no exceptions are logged on that page.
> 
> Did you test this?

It was PluginManager#getTarget 😉 sorted

**Comment by timja** [comment]
OH right, there's exceptions...

**Review by oleg-nenashev — APPROVED** [review]
Awkward part here is that users with `MANAGE` permission will not be able to access the page. Maybe we need new API which allows requiring one-of-multiple permissions.

**Comment by timja** [comment]
> Awkward part here is that users with `MANAGE` permission will not be able to access the page. Maybe we need new API which allows requiring one-of-multiple permissions.

The is an API for that, it was added in the system read PR for /configure,
`permissions` on l:layout

The workaround on /manage was to allow `Overall/Read` to see the managementlink, 

There was a different iteration which added an `isShowManagementLink()` which could be added if needed

**Review by jvz — APPROVED** [review]

**Review by markjacksonfishing — APPROVED** [review]

**Review by jeffret-b — APPROVED** [review]
Looks like it should work fine. Code looks good. I did some cursory testing and it looks like it's behaving fine.

File: core/src/main/java/hudson/PluginManager.java
Comment (daniel-beck) [inline core/src/main/java/hudson/PluginManager.java]: Use the solution in the large Overall/Manage PR instead, please.

**Review by daniel-beck — DISMISSED** [review]

File: core/src/main/java/hudson/PluginManager.java
Comment (timja) [reply core/src/main/java/hudson/PluginManager.java]: Hmm? What are you asking for here?

Comment (timja) [reply core/src/main/java/hudson/PluginManager.java]: done in https://github.com/jenkinsci/jenkins/pull/4520/commits/1a15e159e0e630622679650895f9b0d6b009dc80

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
Added ready-for-merge, there's been quite a lot of review, and half of the security team.

Will merge it tomorrow if no negative feedback

Thanks all

**Pull request merged**
